server/src/main/java/org/elasticsearch/action/admin/indices/shards/IndicesShardStoresResponse.java

```diff
@@ -241,19 +239,15 @@ public IndicesShardStoresResponse(Map<String, Map<Integer, List<StoreStatus>>> s
     }
 
     IndicesShardStoresResponse() {
-        this(ImmutableOpenMap.of(), Collections.emptyList());
+        this(Map.of(), Collections.emptyList());
     }
 
     public IndicesShardStoresResponse(StreamInput in) throws IOException {
         super(in);
-        storeStatuses = in.readImmutableOpenMap(StreamInput::readString, i -> {
-            int indexEntries = i.readVInt();
-            ImmutableOpenIntMap.Builder<List<StoreStatus>> shardEntries = ImmutableOpenIntMap.builder();
-            for (int shardCount = 0; shardCount < indexEntries; shardCount++) {
-                shardEntries.put(i.readInt(), i.readList(StoreStatus::new));
-            }
-            return shardEntries.build();
-        });
+        storeStatuses = in.readImmutableMap(
+            StreamInput::readString,
+            i -> i.readImmutableMap(StreamInput::readInt, j -> j.readList(StoreStatus::new))
```

Comment:
It would be great to have a method to read immutable list as well if possible

Reply:
I opened https://github.com/elastic/elasticsearch/pull/86433